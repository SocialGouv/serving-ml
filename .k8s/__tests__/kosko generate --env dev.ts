//

import { getEnvManifests } from "@socialgouv/kosko-charts/testing";
import { project } from "@socialgouv/kosko-charts/testing/fake/github-actions.env";

jest.setTimeout(1000 * 60);
test("kosko generate --dev", async () => {
  // process.env.HARBOR_PROJECT = "cdtn";
  expect(
    await getEnvManifests("dev", "", {
      ...project("serving-ml").dev,
      RANCHER_PROJECT_ID: "c-bd7z2:p-wcxt4",
      KUBE_NAMESPACE: "serving-ml-144-master-dev2",
    })
  ).toMatchSnapshot();
});
