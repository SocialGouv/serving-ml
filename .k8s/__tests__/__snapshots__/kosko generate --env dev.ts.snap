// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kosko generate --dev 1`] = `
"---
apiVersion: v1
kind: Namespace
metadata:
  annotations:
    field.cattle.io/creatorId: gitlab
    field.cattle.io/projectId: c-bd7z2:p-wcxt4
    git/branch: master
    git/remote: >-
      https://gitlab-ci-token:[MASKED]@gitlab.factory.social.gouv.fr/SocialGouv/serving-ml.git
    app.gitlab.com/app: socialgouv-serving-ml
    app.gitlab.com/env: master-dev2
    app.gitlab.com/env.name: master-dev2
  labels:
    azure-pg-admin-user: serving-ml
    application: master-dev2-serving-ml
    owner: serving-ml
    team: serving-ml
    cert: wildcard
  name: serving-ml-144-master-dev2
---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-serving-ml
    app.gitlab.com/env: master-dev2
    app.gitlab.com/env.name: master-dev2
  labels:
    app: serving-ml-cache
    application: master-dev2-serving-ml
    owner: serving-ml
    team: serving-ml
    cert: wildcard
  name: serving-ml-cache
  namespace: serving-ml-144-master-dev2
spec:
  replicas: 1
  selector:
    matchLabels:
      app: serving-ml-cache
  template:
    metadata:
      annotations:
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        app.gitlab.com/app: socialgouv-serving-ml
        app.gitlab.com/env: master-dev2
        app.gitlab.com/env.name: master-dev2
      labels:
        app: serving-ml-cache
        application: master-dev2-serving-ml
        owner: serving-ml
        team: serving-ml
        cert: wildcard
    spec:
      containers:
        - image: nginx:1.19.6
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /v1/models/sentqam
              port: http
            initialDelaySeconds: 15
            periodSeconds: 5
            timeoutSeconds: 15
          name: serving-ml-cache
          ports:
            - containerPort: 80
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /v1/models/sentqam
              port: http
            initialDelaySeconds: 5
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 3
          resources:
            limits:
              cpu: 1000m
              memory: 1Gi
            requests:
              cpu: 100m
              memory: 0.5Gi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /v1/models/sentqam
              port: http
            periodSeconds: 5
            initialDelaySeconds: 0
            timeoutSeconds: 15
          env:
            - name: UPSTREAM
              value: http://serving-ml
            - name: MAX_SIZE
              value: 1024m
          volumeMounts:
            - mountPath: /var/cache/nginx
              name: cache
            - mountPath: /etc/nginx/nginx.conf
              subPath: nginx.conf
              name: config
          envFrom:
            - secretRef:
                name: cache-sealed-secret
      volumes:
        - name: cache
          azureFile:
            secretName: cache-sealed-secret
            shareName: http-cache
            readOnly: false
        - name: config
          configMap:
            name: nginx-config
---
apiVersion: bitnami.com/v1alpha1
kind: SealedSecret
metadata:
  annotations:
    sealedsecrets.bitnami.com/cluster-wide: 'true'
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-serving-ml
    app.gitlab.com/env: master-dev2
    app.gitlab.com/env.name: master-dev2
  name: cache-sealed-secret
  labels:
    application: master-dev2-serving-ml
    owner: serving-ml
    team: serving-ml
    cert: wildcard
  namespace: serving-ml-144-master-dev2
spec:
  encryptedData:
    azurestorageaccountname: >-
      AgBdaDxvfdZkwR0ELCIYkSRp8App+LJcgNZ6OqnqX7m8VX7eL/zemqw5yCJQa2JjXbSXE3STHT4xmMW7zh+ot50kYAxOsbQDJrAkVzTt3Z5ZFiEJFt6IemXj/Zp2m9Xxp/vo1mtYoGbKcnJvysMkM+4gqSjoHrVRir5+2rLacmnx1p5qu3vQ4BEfmhmt1mICOTaKcOKeJv6OfLRiaCEvZhM9+IEEnSnFz/Exql0Cna/3ZYW9zmYh4ogh4pHGC5JAaVEbgWmI7ByTET5S0yWqTfF1yn62Q+Z8rGKZVCcusL7QjS7SvmXFwM9BTbpYmrdLExaRWZvyslniLKBQv+heoWnkb1HXlxneUaBsjQ5F0kH0/zqhRq0O7e+QJNvuIF3UHzMCm4LSqMAOpidFRPRqJ4j8pFF9FQUa1IvMBDG/mT1Urh7xTIJMQ6yfrPPFTswsEtZ4GHDre/StOLa2EX0sivCoCaQBYRrqK/nYWaCLa07kSwJ5PZ0jkDPmQtm1nEMoFyGT7sbA8vi4qeXnaWK3yvUFgUhj5/T4qxhkZd9jyxLPSzUkNM7jgG8l1dMiRw9g9xH6Zgp4htykthP8yEwOxGfv8tUm/FDATya4Yiy8KAAp5dwAu2Up0d0NVJ5Z8qBTdz1BFRbtsU6twNH35aZVHS+6U/iTlcmfT5AurgErMj8bWpV2BAi0wMleuJBT3dAYshpvM8ydhh577Ds+H1o=
    azurestorageaccountkey: >-
      AgBJuiFrHxkuEyqtasOCBC+hvQpPdrYaCLpZuTamb8Jv9KCANUQbFAFcB7oFBr2wSpGch3t53f5LXqlu25QTCF6GC6t6hgpt+cDxo5pD0UC0Oic1yc7n74bqO96oz8DiUZQb8DlmvCbb7Z+EYeKpGoyqcxWddoBn0i5CJi2zU8q417Y67EJ3bj6fx61ykeEFbuMvPqggR1mpJx+ZfdZ7v+pggxEqTh3IH5/YfYS+Lzk7DC7wMkoZrVDy7ChhbtrZ7Umad1+dM3D8dg2ZmjKvyWZVwffc+RORyb5FancVnw627jGnVNKHXwFKZ1yzoyeU9HI8NmGXJ/YLk4z0jSapw9old0Ujb5ypBkpO+lW36nX58jydI4BaRuz6CbnCxwFE1BfLZHxA7bzYu1VkXDIYaKSv8PGEqyab5PMJt4N4Av+FsMW6xroZTPPMHSUh8HrCzrD0j46v18FLbjejy3tsgc2P1JY0rJl5FEylywnMPNnlL62intmLtm8Xp5F00jDaiq5tLQyMpwaVVaZv4uX3/0n3c1dvIKjRIFv6g/LkL6qY9YUZhOTmVRoIs6N1dwMx17xj/NoPbUC/0uZ9N2AGWan6ZAigkVTrRPU4+W169fG3gciFtMxIuWLCgpAZrNEkvxbzlTqCWoLxzkBfkmIisn6DKPuJrmHw+ZP9shw1PIv01p1Cjke+urmoZSSPEkZhjCGDsxpbIzIWV4SzAkjfZ0VS2Ls3AjfqhG0rt65tRfnubPehut5jUaFZ8qgdl1i9uXov/kFqAn8EQGQM5Zkk2WeCCplnxR5fg6eKyHHelBCNIN4gFUmpnowz
  template:
    metadata:
      annotations:
        sealedsecrets.bitnami.com/cluster-wide: 'true'
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        app.gitlab.com/app: socialgouv-serving-ml
        app.gitlab.com/env: master-dev2
        app.gitlab.com/env.name: master-dev2
      name: cache-sealed-secret
      labels:
        application: master-dev2-serving-ml
        owner: serving-ml
        team: serving-ml
        cert: wildcard
    type: Opaque
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: serving-ml-cache
    application: master-dev2-serving-ml
    owner: serving-ml
    team: serving-ml
    cert: wildcard
  name: serving-ml-cache
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-serving-ml
    app.gitlab.com/env: master-dev2
    app.gitlab.com/env.name: master-dev2
  namespace: serving-ml-144-master-dev2
spec:
  ports:
    - name: http
      port: 80
      targetPort: 80
  selector:
    app: serving-ml-cache
  type: ClusterIP
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-serving-ml
    app.gitlab.com/env: master-dev2
    app.gitlab.com/env.name: master-dev2
  labels:
    app: serving-ml-cache
    application: master-dev2-serving-ml
    owner: serving-ml
    team: serving-ml
    cert: wildcard
  name: serving-ml-cache
  namespace: serving-ml-144-master-dev2
spec:
  rules:
    - host: master-dev2-serving-ml.dev2.fabrique.social.gouv.fr
      http:
        paths:
          - backend:
              serviceName: serving-ml-cache
              servicePort: 80
            path: /
  tls:
    - hosts:
        - master-dev2-serving-ml.dev2.fabrique.social.gouv.fr
      secretName: wildcard-crt
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: nginx-config
data:
  nginx.conf: |

    events {
     worker_connections 1024;
    }

    http {
     include mime.types;
     default_type application/octet-stream;
     sendfile on;
     keepalive_timeout 65;
     client_max_body_size 200k;

     proxy_buffering on;
     proxy_cache_path /cache levels=1:2 keys_zone=small:1m inactive=1w use_temp_path=off;

     server {
      listen 80;
      server_name localhost;

      location / {
       try_files $uri @backend;
      }

      location @backend {
       proxy_cache small;
       proxy_cache_methods POST;
       proxy_pass http://api:80;
       proxy_cache_key \\"$request_uri|$request_body\\";
       proxy_cache_valid any 1w;
       proxy_buffers 8 32k;
       proxy_buffer_size 64k;
       proxy_cache_use_stale updating;
       add_header X-Cached $upstream_cache_status;
      }
     }

    }
---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-serving-ml
    app.gitlab.com/env: master-dev2
    app.gitlab.com/env.name: master-dev2
  labels:
    app: serving-ml
    application: master-dev2-serving-ml
    owner: serving-ml
    team: serving-ml
    cert: wildcard
  name: serving-ml
  namespace: serving-ml-144-master-dev2
spec:
  replicas: 1
  selector:
    matchLabels:
      app: serving-ml
  template:
    metadata:
      annotations:
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        app.gitlab.com/app: socialgouv-serving-ml
        app.gitlab.com/env: master-dev2
        app.gitlab.com/env.name: master-dev2
      labels:
        app: serving-ml
        application: master-dev2-serving-ml
        owner: serving-ml
        team: serving-ml
        cert: wildcard
    spec:
      containers:
        - image: >-
            harbor.fabrique.social.gouv.fr/cdtn/serving-ml:0123456789abcdefghijklmnopqrstuvwxyz0123
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /v1/models/sentqam
              port: http
            initialDelaySeconds: 15
            periodSeconds: 5
            timeoutSeconds: 15
          name: serving-ml
          ports:
            - containerPort: 8501
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /v1/models/sentqam
              port: http
            initialDelaySeconds: 5
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 3
          resources:
            limits:
              cpu: 1000m
              memory: 2Gi
            requests:
              cpu: 500m
              memory: 1.5Gi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /v1/models/sentqam
              port: http
            periodSeconds: 5
            initialDelaySeconds: 0
            timeoutSeconds: 15
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: serving-ml
    application: master-dev2-serving-ml
    owner: serving-ml
    team: serving-ml
    cert: wildcard
  name: serving-ml
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-serving-ml
    app.gitlab.com/env: master-dev2
    app.gitlab.com/env.name: master-dev2
  namespace: serving-ml-144-master-dev2
spec:
  ports:
    - name: http
      port: 80
      targetPort: 8501
  selector:
    app: serving-ml
  type: ClusterIP
"
`;
